Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp

```diff
@@ -0,0 +1,1053 @@
+            static constexpr const char tagName[] = "label";
+        };
+
+        struct Option : ContainerTag<HTMLOptionElement, PermittedParents::Special> {
+            static constexpr const char tagName[] = "option";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                // <option> can only contain a text content.
+                return self.didFail(HTMLFastPathResult::FailedOptionWithChild, nullptr);
+            }
+        };
+
+        struct Ol : ContainerTag<HTMLOListElement, PermittedParents::FlowContent> {
+            static constexpr const char tagName[] = "ol";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                return self.parseSpecificElements<Li>();
+            }
+        };
+
+        struct P : ContainsPhrasingContentTag<HTMLParagraphElement, PermittedParents::FlowContent> {
+            static constexpr const char tagName[] = "p";
+        };
+
+        struct Select : ContainerTag<HTMLSelectElement, PermittedParents::PhrasingOrFlowContent> {
+            static constexpr const char tagName[] = "select";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                return self.parseSpecificElements<Option>();
+            }
+        };
+
+        struct Span : ContainsPhrasingContentTag<HTMLSpanElement, PermittedParents::PhrasingOrFlowContent> {
+            static constexpr const char tagName[] = "span";
+        };
+
+        struct Strong : ContainsPhrasingContentTag<HTMLElement, PermittedParents::PhrasingOrFlowContent> {
+            static constexpr const char tagName[] = "strong";
+
+            static Ref<HTMLElement> create(Document& document)
+            {
+                return HTMLElement::create(HTMLNames::strongTag, document);
+            }
+        };
+
+        struct Ul : ContainerTag<HTMLUListElement, PermittedParents::FlowContent> {
+            static constexpr const char tagName[] = "ul";
+
+            static RefPtr<Element> parseChild(HTMLFastPathParser& self)
+            {
+                return self.parseSpecificElements<Li>();
+            }
+        };
+    };
+
+    template<class ParentTag> void parseCompleteInput()
+    {
+        parseChildren<ParentTag>(m_fragment);
+        if (m_position != m_end)
+            didFail(HTMLFastPathResult::FailedDidntReachEndOfInput);
+    }
+
+    // Match ASCII Whitespace according to https://infra.spec.whatwg.org/#ascii-whitespace
+    bool isWhitespace(Char c)
+    {
+        switch (c) {
+        case ' ':
+        case '\t':
+        case '\n':
+        case '\r':
+        case '\f':
+            return true;
+        default:
+            return false;
+        }
+    }
+
+    bool isValidUnquotedAttributeValueChar(Char c)
+    {
+        return ('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z') || ('0' <= c && c <= '9') || c == '_' || c == '-';
```

Comment:
Seems to include a less optimized `isASCIIAlphanumeric` in here.

Reply:
It seems to allow `_` and '-' which `isASCIIAlphanumeric` wouldn't. We may be able to use `isASCIIAlphanumeric(c) || c == '_' || c == '-'`. Do you mind if I try this in a follow-up. I really should perf test before making such changes.